File: GST_Lesson7_HieuNT94/GST_Lesson7_Ex1_HieuNT94/app/src/main/java/gst/trainingcourse/lesson7_ex1_hieunt94/MainActivity.java
package gst.trainingcourse.lesson7_ex1_hieunt94;

import android.content.Intent;
import android.os.Bundle;

import androidx.fragment.app.FragmentActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

public class MainActivity extends FragmentActivity implements SongListFragment.OnFragmentInteractionListenter {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        SongListFragment songListFragment = new SongListFragment();
        fragmentTransaction.add(R.id.container_songList, songListFragment);
        fragmentTransaction.commit();
    }

    @Override
    public void onSongSelected(Song song) {

        Bundle bundle = new Bundle();
        bundle.putSerializable("song", song);
        if(findViewById(R.id.container_songDetail) != null) {
            SongDetailFragment songDetailFragment = new SongDetailFragment();
            songDetailFragment.setArguments(bundle);

            FragmentManager fragmentManager = getSupportFragmentManager();
            FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
            fragmentTransaction.replace(R.id.container_songDetail, songDetailFragment);
            fragmentTransaction.commit();
        } else {
            Intent intent = new Intent(this, SongDetailActivity.class);
            intent.putExtras(bundle);
            startActivity(intent);
        }
    }
}

File: GST_Lesson6_HieuNT94/app/src/main/java/gst/trainingcourse/lesson6_hieunt94/MainActivity.kt
package gst.trainingcourse.lesson6_hieunt94

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val layoutManager = LinearLayoutManager(this)
        layoutManager.orientation = RecyclerView.VERTICAL
        recycleViewlistSong.layoutManager = layoutManager

        val layoutManagerAuthor = LinearLayoutManager(this)
        layoutManagerAuthor.orientation = LinearLayoutManager.HORIZONTAL
        recycleViewlistAuthor.layoutManager = layoutManagerAuthor


        val adapter = SongAdapter(this, Song.song)
        recycleViewlistSong.adapter = adapter

        val adapterAuthor = AuthorAdapter(this, Song.song)
        recycleViewlistAuthor.adapter = adapterAuthor
    }
}
